src/java.desktop/share/classes/javax/swing/plaf/synth/SynthParser.java

```diff
@@ -213,26 +207,23 @@ class SynthParser extends DefaultHandler {
      *
      * @param inputStream XML document containing the styles to read
      * @param factory DefaultSynthStyleFactory that new styles are added to
-     * @param urlResourceBase the URL used to resolve any resources, such as Images
      * @param classResourceBase the Class used to resolve any resources, such as Images
      * @param defaultsMap Map that UIDefaults properties are placed in
      */
     public void parse(InputStream inputStream,
                       DefaultSynthStyleFactory factory,
-                      URL urlResourceBase, Class<?> classResourceBase,
+                      Class<?> classResourceBase,
                       Map<String, Object> defaultsMap)
                       throws ParseException, IllegalArgumentException {
-        if (inputStream == null || factory == null ||
-            (urlResourceBase == null && classResourceBase == null)) {
+        if (inputStream == null || factory == null || classResourceBase == null) {
             throw new IllegalArgumentException(
-                "You must supply an InputStream, StyleFactory and Class or URL");
+                "You must supply an InputStream, StyleFactory and Class");
         }
 
-        assert(!(urlResourceBase != null && classResourceBase != null));
+        assert(classResourceBase != null);
```

Comment:
It looks like this assert is no longer needed, because the `classResourceBase` is not null at this point.

Reply:
Fair enough. But I'm not sure it was ever needed. The condition in the existing code also ensured it would never fail.

Reply:
fixed

Reply:
> But I'm not sure it was ever needed.

It seems that this was a check that both `urlResourceBase` and `classResourceBase` are not null at the same time, but only when assertions are enabled.
Not sure why it was an assertion.